lib/octokit/repository.rb

```diff
@@ -10,7 +10,10 @@ class Repository
     #
     # @return [Repository]
     def self.from_url(url)
-      Repository.new(URI.parse(url).path[1..-1])
+      new URI.parse(url).path[1..-1].
+        gsub('repos/','').
```

Comment:
This seems unsafe if, e.g., I have a repository named `aroben/all-my-repos/`. I wonder if it would be safer to have a few defined formats we're willing to accept (perhaps defined by regexes) and explicitly matching against each format?